### PR TITLE
neovim: 0.2.2 -> 0.3.0

### PR DESCRIPTION
**Pull request by colemickens**
###### Motivation for this change
v0.3.0 was released today

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by andir** [comment]
@grahamcofborg build neovim

File: pkgs/applications/editors/neovim/default.nix [hunk 1]
```diff
@@ -32,6 +32,7 @@ let
       luaPackages.lua
       gperf
     ] ++ optional withJemalloc jemalloc
+      ++ optional stdenv.isDarwin libiconv
```
Comment (Mic92) [inline pkgs/applications/editors/neovim/default.nix]: This is a blind fix. Let's see if it works.

**Comment by Mic92** [comment]
@GrahamcOfBorg build neovim

**Comment by Mic92** [comment]
aarch64 failed already before that.

**Comment by colemickens** [comment]
Thanks  @Mic92 !

File: pkgs/applications/editors/neovim/default.nix (hunk 1 above)
Comment (matthewbauer) [reply pkgs/applications/editors/neovim/default.nix]: Note that we almost always want to unconditionally add libiconv. Things like musl will also need it as well.

**Pull request merged**